### PR TITLE
fixed few problems that caused errno to be lost

### PR DESCRIPTION
**Pull request by crusader-mike**
fix for issues mentioned in #1756

### CODE REVIEW

File: src/unix/core.c [hunk 1]
```diff
@@ -536,7 +536,7 @@ int uv__close_nocheckstdio(int fd) {
 int uv__close(int fd) {
   assert(fd > STDERR_FILENO);  /* Catch stdio close bugs. */
 #if defined(__MVS__)
-  epoll_file_close(fd);
+  SAVE_ERRNO( epoll_file_close(fd) );
```
Comment (bnoordhuis) [inline src/unix/core.c]: Style: no spaces.

**Comment by crusader-mike** [comment]
pardon my ignorance, but:
- how can change log message? and how it should look like?
- how can I add another change to this pull request?
- should I simply start a new pull request instead?
- should I squash my commits (if there is more than one in my change)?

**Comment by davisjam** [comment]
Hi @crusader-mike 

Answering each of your questions, I hope:

> how can change log message?

`git commit --amend` will let you edit the commit message at HEAD. For multiple commits you can use `git rebase -i` and mark any commits you with to reword with 'r'.

> and how it should look like?

There's a description of the preferred style [here](https://github.com/libuv/libuv/blob/v1.x/CONTRIBUTING.md#commit). You can run `git log` and see what they usually look like, too.

> how can I add another change to this pull request?

Just run `git commit` and then `git push` while in the same branch. GitHub will detect and update this PR. If you are updating the commit you can run `git add changed-file1 changed-file2 ...; git commit --amend` which will modify the original commit. Then 'git push --force' will re-write the remote branch (the one we can see) to have only the new commit. This is "rewriting history" but it's OK as long as the PR branch hasn't been merged.

> should I simply start a new pull request instead?

No. Then we have to look at two PRs instead of one.

> should I squash my commits (if there is more than one in my change)?

Write "good commits" -- one logical thing. If it takes more than one commit to accomplish the goal, that's OK. If in doubt don't bother and we can discuss later.

**Comment by crusader-mike** [comment]
@davisjam Done. Not sure if I was supposed to add "Reviewed-By" to commit message. Kind of hard to know who is a reviewer before creating a PR.

**Comment by davisjam** [comment]
> Not sure if I was supposed to add "Reviewed-By" to commit message

Nope. That gets added by the person merging the PR.

**Comment by cjihrig** [comment]
Landed in https://github.com/libuv/libuv/commit/9ed3ed5fcb3f19eccd3d29848ae2ff0cfd577de9. Thanks!

**Pull request closed without merging**